src/dash/DashHandler.js

```diff
@@ -283,6 +283,10 @@ function DashHandler(config) {
 
         let indexToRequest = lastSegment ? lastSegment.index + 1 : 0;
 
+        if (representation.endNumber && indexToRequest >= representation.endNumber) {
```

Comment:
The tests for the `DashHandler` file are a bit weird, there are a couple of tests that only test for the `null` case and not for the `success` case. Because of that, the setup needs to be changed if we actually want to test things. 

Should I do that?

Reply:
If you can address that as well it would be great